src/backend/distributed/commands/table.c

```diff
@@ -507,6 +508,12 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand,
 		return PreprocessAlterTableStmt(node, alterTableCommand, processUtilityContext);
 	}
 
+	if (DropsForeignKey(leftRelationId, alterTableStatement))
```

Comment:
I was expecting this check to be in the case where `referencingIsLocalTable` is local, all other cases should have been already covered:
```C
if (referencingIsLocalTable)
{
  if (DropsForeignKey(leftRelationId, alterTableStatement))
  {
      invalidate;
   }
return;
}
```


> /* invalidating the fkey graph since cached foreign keys could cause crashes */

Also, we can maybe improve the wording a bit:
```
The foreign key graph keeps track of the foreign keys including local tables. So, even if a foreign key on a local table is dropped, we should invalidate the graph so that the next commands  can see the graph up-to date
```